packages/error-view/__tests__/use-error-page-analytics.test.tsx

```diff
@@ -0,0 +1,26 @@
+import { render, waitFor } from '@testing-library/react'
+import useErrorPageAnalytics from '../use-error-page-analytics'
+
+function HookTestComponent({ statusCode }) {
+  useErrorPageAnalytics(statusCode)
+  return <div>Testing the useErrorPageAnalytics hook...</div>
+}
+
+describe('useErrorPageAnalytics', () => {
+  it('calls window.analytics.track with the provided error code', async () => {
+    // Mock window.analytics
+    const forMockRestore = window.analytics
+    // $TSFixMe loosens window.analytics type to diverge from Segment type
+    // defined in @hashicorp/platform-types
+    window.analytics = { track: jest.fn() } as $TSFixMe
```

Comment:
Without the `... as $TSFixMe` I'd end up with TypeScript getting mad that this doesn't match the type defined in https://github.com/hashicorp/web-platform-packages/blob/6062939845d5d841d78afade8225e891cd428aa3/packages/types/index.d.ts#L34. Hopefully this makes sense, but wanted to flag in case anyone has a better workaround.